qa/L0_shared_memory/shared_memory_test.py

```diff
@@ -29,124 +29,143 @@
 import sys
 
 sys.path.append("../common")
-
 import os
 import unittest
+from pathlib import Path
 
 import numpy as np
 import test_util as tu
 import tritonclient.grpc as grpcclient
 import tritonclient.http as httpclient
 import tritonclient.utils.shared_memory as shm
+import util
+from parameterized import parameterized
 from tritonclient import utils
 
 
 class SharedMemoryTest(tu.TestResultCollector):
-    def test_invalid_create_shm(self):
-        # Raises error since tried to create invalid system shared memory region
-        try:
-            shm_op0_handle = shm.create_shared_memory_region(
-                "dummy_data", "/dummy_data", -1
-            )
-            shm.destroy_shared_memory_region(shm_op0_handle)
-        except Exception as ex:
-            self.assertTrue(str(ex) == "unable to initialize the size")
+    # Constant members
+    shared_memory_test_client_log = Path(os.getcwd()) / "client.log"
+    model_dir_path = Path(os.getcwd()) / "models"
+    model_source_path = Path(os.getcwd()).parents[0] / "python_models/add_sub/model.py"
+    model_config_source_path = (
+        Path(os.getcwd()).parents[0] / "python_models/add_sub/config.pbtxt"
+    )
 
-    def test_valid_create_set_register(self):
-        # Create a valid system shared memory region, fill data in it and register
-        if _protocol == "http":
-            triton_client = httpclient.InferenceServerClient(_url, verbose=True)
+    # Custom setup method to allow passing of parameters
+    def _setUp(self, protocol, log_file_path):
+        self._tritonserver_ipaddr = os.environ.get("TRITONSERVER_IPADDR", "localhost")
+        self._test_windows = bool(int(os.environ.get("TEST_WINDOWS", 0)))
+        self._shm_key_prefix = "/" if not self._test_windows else "Global\\"
+        self._timeout = os.environ.get("SERVER_TIMEOUT", 120)
+        self._protocol = protocol
+        self._test_passed = False
+        self._original_stdout, self._original_stderr = None, None
+        self._log_file_path = log_file_path
+
+        if self._protocol == "http":
+            self._url = self._tritonserver_ipaddr + ":8000"
+            self._triton_client = httpclient.InferenceServerClient(
+                self._url, verbose=True
+            )
         else:
-            triton_client = grpcclient.InferenceServerClient(_url, verbose=True)
-        shm_op0_handle = shm.create_shared_memory_region("dummy_data", "/dummy_data", 8)
-        shm.set_shared_memory_region(
-            shm_op0_handle, [np.array([1, 2], dtype=np.float32)]
+            self._url = self._tritonserver_ipaddr + ":8001"
+            self._triton_client = grpcclient.InferenceServerClient(
+                self._url, verbose=True
+            )
+        self._build_model_repo()
+        self._build_server_args()
+        self._shared_memory_test_server_log = open(log_file_path, "w")
+        self._server_process = util.run_server(
```

Comment:
How does `util.run_server` interact with `test.sh` also starting server? Is there conflict or issue there?

Reply:
I don't believe they should overlap. For this test my ultimate goal is to remove `test.sh` entirely.

Reply:
Isn't this also getting run in the linux case that runs `test.sh`? or is there changes on the gitlab-side to not run `test.sh` at all?

Reply:
There are changes such that `test.sh` will not run at all.